build.zig: Pkg.dependencies woes

This is on Windows 10. The following repro code works with Zig 0.5.0+ab0576667:

```Zig
const std = @import("std");
const Builder = std.build.Builder;
const Pkg = std.build.Pkg;

pub fn build(b: *Builder) void {
    // Standard target options allows the person running `zig build` to choose
    // what target to build for. Here we do not override the defaults, which
    // means any target is allowed, and the default is native. Other options
    // for restricting supported target set are available.
    const target = b.standardTargetOptions(.{});

    // Standard release options allow the person running `zig build` to select
    // between Debug, ReleaseSafe, ReleaseFast, and ReleaseSmall.
    const mode = b.standardReleaseOptions();

    const exe = b.addExecutable("app", "src/main.zig");
    exe.setTarget(target);
    exe.setBuildMode(mode);
    exe.install();

    const lib1 = Pkg{
        .name = "lib1",
        .path = "../lib1/src/main.zig",
    };
    exe.addPackage(lib1);

    exe.addPackage(Pkg{
        .name = "lib2",
        .path = "../lib2/src/main.zig",
        .dependencies = &[_]Pkg{lib1},
    });

    const run_cmd = exe.run();
    run_cmd.step.dependOn(b.getInstallStep());

    const run_step = b.step("run", "Run the app");
    run_step.dependOn(&run_cmd.step);
}
```

With Zig 0.6.0+c22b54c9d and others in the 0.6 line, the compiler complains:

```C:\devel\Zig\zig-windows-x86_64-0.6.0+c22b54c9d\zig.exe build
Semantic Analysis [1551/2469] .\build.zig:30:32: error: expected type '[]std.build.Pkg', found '*const [1]std.build.Pkg'
        .dependencies = &[_]Pkg{lib1},
                               ^
```

Trying to fix the problem like this:

```Zig
...
    var lib2_deps: [1]Pkg = [_]Pkg{lib1};

    exe.addPackage(Pkg{
        .name = "lib2",
        .path = "../lib2/src/main.zig",
        .dependencies = &lib2_deps,
    });
...
```

results in:

```
C:\devel\Zig\zig-windows-x86_64-0.6.0+c22b54c9d\zig.exe build
Segmentation fault at address 0x1
C:\devel\Zig\zig-windows-x86_64-0.6.0+c22b54c9d\lib\zig\std\fs\path.zig:248:35: 0x7ff7dfaf0a4d in std.fs.path.windowsParsePath (build.obj)
    if (path.len >= 2 and path[1] == ':') {
                                  ^
```

--- a/lib/std/build.zig
+++ b/lib/std/build.zig
@@ -1054,7 +1054,7 @@ pub const Target = std.zig.CrossTarget;
 pub const Pkg = struct {
     name: []const u8,
     path: []const u8,
-    dependencies: ?[]Pkg = null,
+    dependencies: ?[]const Pkg = null,
 };
 
 const CSourceFile = struct {

